src/frontend/src/handler/create_sink.rs

```diff
@@ -180,9 +218,107 @@ pub fn gen_sink_plan(
         dependent_relations.into_iter().collect_vec(),
     );
 
-    Ok((query, sink_plan, sink_catalog))
+    if let Some(table_catalog) = &target_table_catalog {
+        for column in sink_catalog.full_columns() {
+            if column.is_generated() {
+                return Err(RwError::from(ErrorCode::BindError(
+                    "The sink to table feature for Sinks with generated columns has not been implemented yet."
+                        .to_string(),
+                )));
+            }
+        }
+
+        let user_defined_primary_key_table =
+            !(table_catalog.append_only || table_catalog.row_id_index.is_some());
+
+        if !(user_defined_primary_key_table
+            || sink_catalog.sink_type == SinkType::AppendOnly
+            || sink_catalog.sink_type == SinkType::ForceAppendOnly)
+        {
+            return Err(RwError::from(ErrorCode::BindError(
+                "Only append-only sinks can sink to a table without primary keys.".to_string(),
+            )));
+        }
+
+        let exprs = derive_default_column_project_for_sink(&sink_catalog, table_catalog)?;
+
+        let logical_project = generic::Project::new(exprs, sink_plan);
+
+        sink_plan = StreamProject::new(logical_project).into();
+
+        let exprs =
+            LogicalSource::derive_output_exprs_from_generated_columns(table_catalog.columns())?;
+        if let Some(exprs) = exprs {
+            let logical_project = generic::Project::new(exprs, sink_plan);
+            sink_plan = StreamProject::new(logical_project).into();
+        }
+    };
+
+    Ok((query, sink_plan, sink_catalog, target_table_catalog))
 }
 
+fn check_cycle_for_sink(
```

Comment:
We might need to check it on the meta because https://github.com/risingwavelabs/risingwave/issues/5895?
or could it be rejected by other mechanisms?